remo/domain/dataset.py

```diff
@@ -292,13 +297,17 @@ def get_annotation_set(self, annotation_set_id: int = None) -> AnnotationSet:
         if not annotation_set_id:
             return self.default_annotation_set()
 
-        annotation_set = self.sdk.get_annotation_set(id)
-        if annotation_set.dataset_id == self.id:
+        annotation_set = self.sdk.get_annotation_set(annotation_set_id)
+        if (annotation_set) & (annotation_set.dataset_id == self.id):
```

Comment:
```suggestion
        if annotation_set and annotation_set.dataset_id == self.id:
```

`&` in python is binary operator, applied to bits